### PR TITLE
Add array serializer and test

### PR DESCRIPTION
**Pull request by joheredi**
Moving serialize array to its own file and add tests

### CODE REVIEW

File: packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+      expect(result).to.equal("myArray.map(Date)");
+    });
+
+    it("should handle a model array", async () => {
+      const { modelArray } = testArrays;
+      const result = serializeArray(modelArray);
+
+      const expected = await format(
+        `myArray.map((e: Foo)=>({...(e),
```
Comment (joheredi) [inline packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts:144]: Serializer and deserializer will have (e: Foo) however one refers to the Modular Foo and the other to the RLC foo. @dgetu are we handling this already or this needs a fix

Comment (joheredi) [reply packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts:180]: I was not setting up the serializerMap that's why I was getting the same name for RLC and Modular references

File: packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+      const actual = await format(result, { parser: "typescript" });
+
+      expect(actual).to.equal(expected);
+    });
+
+    it("should handle a model array ", async () => {
+      const { modelArray } = testArrays;
+      const result = serializeArray(modelArray);
+
+      const expected = await format(`myArray.map((e) => serializeFoo(e));;`, {
```
Comment (dgetu) [inline packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts:79]: Will the formatter catch this duplicated semicolon? Or is that expected?

**Review by dgetu — APPROVED** [review]
Overall looks good! I'm not too familiar with the header requirements, so I'll follow up if we run into validation issues.

File: packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts (hunk 2 above)
Comment (joheredi) [reply packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts:79]: Yeah it is a typo on my side, formatter will remove it. I'll fix it in the next commit

**Pull request merged**